overlord/devicestate/devicemgr.go

```diff
@@ -885,10 +885,30 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 assume we are in the run mode
-	if sysAction.Mode == "run" {
-		// do nothing
-		return nil
+	switch m.systemMode {
+	case "recover", "run":
+		// if going from recover to recover or from run to run and the systems
+		// are the same do nothing
+		if m.systemMode == sysAction.Mode && systemLabel == currentSys.System {
+			return nil
+		}
+	case "install":
+		// unclear why we should support requesting the system to go into
+		// install mode when it's already in install mode, but sure why not?
+		if sysAction.Mode == "install" && systemLabel == currentSys.System {
```

Comment:
currentSys will be null if this early in an install, vs reinstall.  These cases need to be tested